jedi.inc

```diff
@@ -609,21 +609,31 @@
   HAS_UNIT_PNGIMAGE              Unit PngImage exists (D12+)
   HAS_UNIT_CHARACTER             Unit Character exists (D12+)
   XPLATFORM_RTL                  The RTL supports crossplatform function names (e.g. RaiseLastOSError) (D6+/BCB6+/FPC)
+  SUPPORTS_POINTERMATH           Compiler supports arithmetic operations on typed pointers (D12+)
+  SUPPORTS_STRINGCHECKS          Compiler supports the STRINGCHECKS directive (D12+) (automatic payload-checks using Assure*String functions, relevant for C++ Builder)
   SUPPORTS_UNICODE               string type is aliased to an unicode string (WideString or UnicodeString) (DX.NET, D12+)
   SUPPORTS_UNICODE_STRING        Compiler supports UnicodeString (D12+)
   SUPPORTS_INT_ALIASES           Types Int8, Int16, Int32, UInt8, UInt16 and UInt32 are defined in the unit System (D12+)
   HAS_UNIT_RTTI                  Unit RTTI is available (D14+)
+  SUPPORTS_EXTENDED_RTTI         Compiler supports the (extended) RTTI directive (D14+)
   SUPPORTS_CAST_INTERFACE_TO_OBJ The compiler supports casts from interfaces to objects (D14+)
   SUPPORTS_DELAYED_LOADING       The compiler generates stubs for delaying imported function loads (D14+)
   HAS_UNIT_REGULAREXPRESSIONSAPI Unit RegularExpressionsAPI is available (D15+)
+  HAS_UNITSCOPE                  Has unit scope (D16+)
   HAS_UNIT_SYSTEM_UITYPES        Unit System.UITypes is available (D16+)
+  HAS_UNIT_ZIP                   Unit Zip exists (D16+)
+  HAS_UNIT_ZLIB                  Unit ZLib exists (D16+)
   HAS_UNIT_SYSTEM_ACTIONS        Unit System.Actions is available (D17+)
   DEPRECATED_SYSUTILS_ANSISTRINGS  AnsiString functions from SysUtils are deprecated and moved to System.AnsiStrings (D18+)
   HAS_PROPERTY_STYLEELEMENTS     TControl has a StyleElements property (D17+)
   HAS_AUTOMATIC_DB_FIELDS        Database fields are automatically created/refreshed (D20+)
   HAS_EARGUMENTEXCEPTION         Exception class EArgumentException is available (D14+)
   HAS_ENOTIMPLEMENTED            Exception class ENotImplemented is available (D15+)
   HAS_UNIT_VCL_THEMES            Unit Vcl.Themes is available (D16+)
+  USE_64BIT_TYPES                Use 64 bit types (D16+)
```

Comment:
Yeah, I don't know what it was for either - it appears below `{$IFDEF COMPILER16_UP}` but wasn't mentioned in the Feature Directives list, hence the added mention now.

This define was introduced here : https://github.com/project-jedi/jedi/commit/67f657a6#diff-c35e1e48c9a0c52bbdb00bbab615f2bf340300851c21b3f693974c72aeb91847R1271

Reply:
I'd simply leave it as is, I mean, do not document it.

Reply:
I've verified that all other defined Feature Directives _are_ documented, and onlyc is lacking a description.

Removing the define itself might break software that depends on it (so that's not an option), thus my reasoning was to at least complete the description list by mentioning `USE_64BIT_TYPES` as well.

I'll do some more digging, perhaps I can find the meaning of this...

Reply:
@outchy Do you by any chance remember for what intented reason you defined `USE_64BIT_TYPES`?

@obones Renaming an existing Feature Directive (like in this case, renaming `USE_64BIT_TYPES` into `SUPPORTS_64BIT_TYPES`) would introduce backwards-compatibility issues, but it _could_ be renamed without consequences when we extend it with an additional explicit backwards-compatible define...

Reply:
When there's no response in the upcoming days, I'll remove this line to resolve this final comment, so that the pull request can be merged.

Reply:
In the mean time, I've updated the pull request description (as a copy of all relevant commit messages, updated with latest decision on how to name the newly introduced directives)

Reply:
Replaced it by a TODO.